### DISCUSSION TITLE
Confused about Thunderbolt firmware version for patching

### DISCUSSION THREAD

**Discussion by dotslashcow** [osy86/HaC-Mini#1020] [Q&A]
According to the [technical details page](https://osy.gitbook.io/hac-mini-guide/details/thunderbolt-3-fix-part-3), the name of the Thunderbolt firmware file osy used is `Intel_Hades_AR_HR_4C_C0_rev33_W_TI_20180109_SEC1_sign.bin`. This seems to match the warning in the Thunderbolt patcher application itself, which says that the patch is only tested with v33.

However, on the Intel website, the latest is v82, with the note _"The Thunderbolt™ 3 Firmware version, for the Intel® NUC Kit NUC8i7HNK and NUC8i7HVK, has not been updated. It was mistakenly listed as version 33. The download file has always been version 82."_

When I dowload the firmware update utility from Intel, the filename of the firmware is `Intel_Hades_AR_HR_4C_C0_rev46_TI_20180109_SEC3_sign`. 

So does this mean that firmware version 82, version 33 and rev46 are all the same? Or am I missing something? Is there any place I can download the v33 firmware?

(I'm currently on v28, running macOS 12.5.1 and HaC-Mini 3.1.5)

Bonus question: Has anyone tried the patcher application on Monterey? (See [this comment](https://github.com/osy/ThunderboltPatcher/issues/9#issuecomment-1227516884) by @osy)

Thanks!

**Comment by ivan-ef** [comment]
> According to the [technical details page](https://osy.gitbook.io/hac-mini-guide/details/thunderbolt-3-fix-part-3), the name of the Thunderbolt firmware file osy used is `Intel_Hades_AR_HR_4C_C0_rev33_W_TI_20180109_SEC1_sign.bin`. This seems to match the warning in the Thunderbolt patcher application itself, which says that the patch is only tested with v33.
> 
> However, on the Intel website, the latest is v82, with the note _"The Thunderbolt™ 3 Firmware version, for the Intel® NUC Kit NUC8i7HNK and NUC8i7HVK, has not been updated. It was mistakenly listed as version 33. The download file has always been version 82."_
> 
> When I dowload the firmware update utility from Intel, the filename of the firmware is `Intel_Hades_AR_HR_4C_C0_rev46_TI_20180109_SEC3_sign`.
> 
> So does this mean that firmware version 82, version 33 and rev46 are all the same? Or am I missing something? Is there any place I can download the v33 firmware?
> 
> (I'm currently on v28, running macOS 12.5.1 and HaC-Mini 3.1.5)
> 
> Bonus question: Has anyone tried the patcher application on Monterey? (See [this comment](https://github.com/osy/ThunderboltPatcher/issues/9#issuecomment-1227516884) by @osy)
> 
> Thanks!

I'm tried thunderbolt patcher in Monterey several times, install and uninstall, no issues at all.

**Reply by dotslashcow** [reply]
Thanks, it does indeed appear to work. I have a lot of strange behaviour after patching, though. (lots of kernel panics, and the best part: after booting and shutting down Linux, the NUC is completely dead until I pull the plug, wait for a minute, and plug it back in). I need to do some more investigation...

**Reply by CuriousRubick** [reply]
Same issues. If anyone figures out how to reflash back to the original firmware, please post here. 

*edit*
I am unable to flash any firmware in windows as intel's official patcher no longer recognizes my device. I cannot reinstall hackintosh to attempt a reflash because only windows usb installers are recognized.

**Comment by CuriousRubick** [comment]
I manage to reinstall mac to my canyon. But I can't figure out how to flash the original firmware. The download from intel linked below includes a .bin file. Can I use that with the thunderbolt patcher?

https://www.intel.com/content/www/us/en/download/19385/thunderbolt-3-firmware-update-tool-for-nuc8i7hnk-nuc8i7hvk.html

**Reply by osy** [reply]
You can reboot into windows via OpenCore which will now recognize the TB3 controller and you can try Intel’s flasher.

**Reply by ivan-ef** [reply]
> You can reboot into windows via OpenCore which will now recognize the TB3 controller and you can try Intel’s flasher.

Just wonder how does thunderbolt works in bootcamp on real Mac with Apple thunderbolt firmware?

**Reply by Medizinmann-MD** [reply]
Hello,
can anyone provide the latest thunderbolt firmware/update tool  (AFAIK v82 was the last one) for the NUC8i7HxK as it is nowhere to be found on the pages from ASUS.

Thanks and best regards.
